log/noop/noop_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package noop // import "go.opentelemetry.io/otel/log/noop"
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+func TestImplementationNoPanics(t *testing.T) {
+	// Check that if type has an embedded interface and that interface has
```

Comment:
> Check that if type has an embedded interface

Where is this check? _I failed to find it in a few minutes so I think it is easier to ask._

Also, isn't the

> // Compile-time check this implements the OpenTelemetry API. 

good enough?

Reply:
If the `embedded` interface is replaced with `log.*` the compile check will not catch the change. This will. 

This also matches out other testing:

- https://github.com/open-telemetry/opentelemetry-go/blob/main/metric/noop/noop_test.go
- https://github.com/open-telemetry/opentelemetry-go/blob/main/trace/noop/noop_test.go

Reply:
The issue was with my understanding of the comment 😅